libs/net/dal/Services/UserService.cs

```diff
@@ -146,5 +146,18 @@ public override User UpdateAndSave(User entity)
 
         throw new InvalidOperationException("User does not exist");
     }
+
+    public IEnumerable<User> FindByRoles(IEnumerable<string> roles)
+    {
+        var query = Context.Users.AsNoTracking();
+        var result = roles.Any() ? Enumerable.Empty<User>() : query;
+
+        foreach (var role in roles)
+        {
+            result = result.Union(query.Where(c => EF.Functions.Like(c.Roles.ToLower(), $"%[{role.ToLower()}]%")));
```

Comment:
This query will result in duplicates if the user belong to more than one role and the function requests has more than one role.
The `User.Roles` property is simple a comma separated string value.  You can instead use the loop to add an Or statement.

Reply:
Updated, thanks, and it seems we could use UnionBy to simplify things a little bit.